test-fixtures/src/main/kotlin/com/alecstrong/sql/psi/test/fixtures/TestHeadlessParser.kt

```diff
@@ -7,22 +7,25 @@ import com.alecstrong.sql.psi.core.SqlFileBase
 import com.alecstrong.sql.psi.core.SqlParserDefinition
 import com.intellij.icons.AllIcons
 import com.intellij.lang.Language
+import com.intellij.lang.LanguageParserDefinitions
 import com.intellij.openapi.fileTypes.LanguageFileType
 import com.intellij.psi.FileViewProvider
 import com.intellij.psi.tree.IFileElementType
 import java.io.File
 
-class TestHeadlessParser {
-  private val parserDefinition = TestParserDefinition()
-
+object TestHeadlessParser {
   fun build(root: String, annotator: SqlAnnotationHolder, predefinedTables: List<PredefinedTable>): SqlCoreEnvironment {
+    val parserDefinition = TestParserDefinition(predefinedTables)
+
     val environment = object : SqlCoreEnvironment(
       sourceFolders = listOf(File(root)),
       dependencies = emptyList(),
-      predefinedTables = predefinedTables,
-      language = parserDefinition.getLanguage(),
     ) {
       init {
+        updateApplication {
+          LanguageParserDefinitions.INSTANCE.removeExplicitExtension(parserDefinition.getLanguage(), LanguageParserDefinitions.INSTANCE.forLanguage(parserDefinition.getLanguage()))
```

Comment:
We need to update the new parser definition to get the new system tables. Otherwise the old parser is used without the updated content.